File: problems/intermediate/organizing_container/organizing_container.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

/**
 * Determines if it is possible to organize containers so that each contains balls of only one type.
 *
 * @param container A 2D vector where container[i][j] is the number of balls of type j in container i
 * @return "Possible" if reorganization is possible, otherwise "Impossible"
 */

string organizingContainers(vector<vector<int>> container) {
     int n = container.size();
    vector<long long> rowSum(n, 0), colSum(n, 0);

    for (int i = 0; i < n; ++i)
        for (int j = 0; j < n; ++j) {
            rowSum[i] += container[i][j];
            colSum[j] += container[i][j];
        }

    sort(rowSum.begin(), rowSum.end());
    sort(colSum.begin(), colSum.end());

    return rowSum == colSum ? "Possible" : "Impossible";
}

File: problems/intermediate/organizing_container/OrganizingContainer.java
package problems.intermediate.organizing_container;

import java.util.*;

public class OrganizingContainer {
    /**
     * Determines if it is possible to organize containers so each contains balls of only one type.
     *
     * @param container 2D int array where container[i][j] is the number of balls of type j in container i
     * @return 'Possible' if reorganization is possible, otherwise 'Impossible'
     */
    public static String organizingContainers(List<List<Integer>> container) {
        int n = container.size();
        List<Long> rowSum = new ArrayList<>(Collections.nCopies(n, 0L));
        List<Long> colSum = new ArrayList<>(Collections.nCopies(n, 0L));

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                int val = container.get(i).get(j);
                rowSum.set(i, rowSum.get(i) + val);
                colSum.set(j, colSum.get(j) + val);
            }
        }

        Collections.sort(rowSum);
        Collections.sort(colSum);

        return rowSum.equals(colSum) ? "Possible" : "Impossible";
    }
}
